test/e2e/subscription_e2e_test.go

```diff
@@ -1366,6 +1366,175 @@ var _ = Describe("Subscription", func() {
+				[]registry.PackageManifest{packageFA, packageC, packageD},
+				[]apiextensions.CustomResourceDefinition{crdA, crdC, crdD},
+				csvs2)
+			catsrc2.Spec.Priority = 100
+
+			catsrc3, c3 = createInternalCatalogSource(kubeClient, crClient, genName("catsrc3"), differentNS,
+				[]registry.PackageManifest{packageFB}, []apiextensions.CustomResourceDefinition{crdB}, csvs3)
+			catsrc3.Spec.Priority = 200
+
+			catsrc4, c4 = createInternalCatalogSource(kubeClient, crClient, genName("catsrc4"), testNamespace,
+				[]registry.PackageManifest{packageFD, packageFC, packageB},
+				[]apiextensions.CustomResourceDefinition{crdB, crdC, crdD}, csvs4)
+			catsrc4.Spec.Priority = 100
+
+			cleanup = func() {
+				c1()
+				c2()
+				c3()
+				c4()
+			}
+		})
+		defer cleanup()
+
+		By("waiting for catalogsources to be ready", func() {
+			_, err := fetchCatalogSourceOnStatus(crClient, catsrc1.GetName(), testNamespace, catalogSourceRegistryPodSynced)
+			Expect(err).ToNot(HaveOccurred())
+			_, err = fetchCatalogSourceOnStatus(crClient, catsrc2.GetName(), testNamespace, catalogSourceRegistryPodSynced)
+			Expect(err).ToNot(HaveOccurred())
+			_, err = fetchCatalogSourceOnStatus(crClient, catsrc3.GetName(), differentNS, catalogSourceRegistryPodSynced)
+			Expect(err).ToNot(HaveOccurred())
+			_, err = fetchCatalogSourceOnStatus(crClient, catsrc4.GetName(), testNamespace, catalogSourceRegistryPodSynced)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		// Create a subscription for packageA in catsrc
+		subscriptionSpec := &v1alpha1.SubscriptionSpec{
+			CatalogSource:          catsrc1.GetName(),
+			CatalogSourceNamespace: catsrc1.GetNamespace(),
+			Package:                packageMain.PackageName,
+			Channel:                stableChannel,
+			InstallPlanApproval:    v1alpha1.ApprovalAutomatic,
+		}
+		subscriptionName := genName("sub-")
+		cleanupSubscription := createSubscriptionForCatalogWithSpec(GinkgoT(), crClient, testNamespace, subscriptionName, subscriptionSpec)
+		defer cleanupSubscription()
+
+		subscription, err := fetchSubscription(crClient, testNamespace, subscriptionName, subscriptionHasInstallPlanChecker)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(subscription).ToNot(BeNil())
+
+		_, err = fetchCSV(crClient, "csv-main", testNamespace, csvSucceededChecker)
+		Expect(err).ToNot(HaveOccurred())
+
+		// ensure correct CSVs were picked
+		var got []string
+		Eventually(func() []string {
+			ip, err := crClient.OperatorsV1alpha1().InstallPlans(testNamespace).Get(context.TODO(), subscription.Status.InstallPlanRef.Name, metav1.GetOptions{})
+			if err != nil {
+				return nil
+			}
+			got = ip.Spec.ClusterServiceVersionNames
+			return got
+		}).ShouldNot(BeNil())
+		Expect(got).Should(ConsistOf([]string{"csv-main", "csv-a", "csv-b", "csv-c", "csv-d"}))
+		//require.ElementsMatch(GinkgoT(), []string{"csv-main", "csv-a", "csv-b", "csv-c", "csv-d"}, got)
```

Comment:
```suggestion
		Eventually(func() ([]string, error) {
			ip, err := crClient.OperatorsV1alpha1().InstallPlans(testNamespace).Get(context.TODO(), subscription.Status.InstallPlanRef.Name, metav1.GetOptions{})
			if err != nil {
				return nil, err
			}
			return ip.Spec.ClusterServiceVersionNames, nil
		}).Should(ConsistOf([]string{"csv-main", "csv-a", "csv-b", "csv-c", "csv-d"}))
```

No need for the intermediate variable `got` here, since the assertion can match the return value directly. Also, including the error as a second return value will make failures more useful when the assertion times out with an error (http://onsi.github.io/gomega/#handling-errors).

Reply:
nice